app/models/resource.py

```diff
@@ -98,8 +99,28 @@ class Resource(db.Model):
         back_populates='resource',
         cascade='save-update, merge, delete, delete-orphan'
     )
-    suggestions = db.relationship('Suggestion', backref='resource',
-                                  uselist=True)
+    type = db.Column(db.String(20))
+
+    __mapper_args__ = {
+        'polymorphic_on': type,
+        'polymorphic_identity': 'resource_base'
+    }
+
+    def __repr__(self):
+        return '<ResourceBase \'%s\'>' % self.name
+
+
+class Resource(ResourceBase):
+    """
+    Schema for resources with relationships to descriptors.
+    """
+    suggestions = db.relationship('ResourceSuggestion', backref='resource',
+                                  uselist=True,
+                                  remote_side='ResourceSuggestion.id')
+
+    __mapper_args__ = {
```

Comment:
the style here is to put __mapper_args__ as the first line in the class: http://docs.sqlalchemy.org/en/latest/orm/extensions/declarative/inheritance.html#single-table-inheritance